src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -183,13 +203,52 @@ private SymbolicRegexMatcher(SymbolicRegexBuilder<TSet> builder, SymbolicRegexNo
                 _positionKinds[mintermId + 1] = CalculateMintermIdKind(mintermId);
             }
 
+            // Create optimized reversal
+            _optimizedReversalState = CreateOptimizedReversal(_pattern.Reverse(builder));
+
             // Store the find optimizations that can be used to jump ahead to the next possible starting location.
             // If there's a leading beginning anchor, the find optimizations are unnecessary on top of the DFA's
             // handling for beginning anchors.
             if (findOptimizations.IsUseful &&
                 findOptimizations.LeadingAnchor is not RegexNodeKind.Beginning)
             {
-                _findOpts = findOptimizations;
+                // this makes some assumptions about the frequency of occurrences
+                // some large sets like \p{Sm} are faster with infrequent matches but slower with frequent matches
+                // the easiest thing to do here is to leave it as-is, but this means some inputs can have large performance losses of 10x or more
```

Comment:
> but this means some inputs can have large performance losses of 10x or more

Isn't that true in the reverse direction as well? e.g. if you disable vectorization searching for > 4 characters but then it turns out the characters being searched for are rare?

Reply:
This is true, but i think for most cases reversal should not be vectorized at all. Reversal vectorization only brings noticeable benefit if the matches are both rare and long (200+ chars on average or so) and even then it's competing with raw DFA speed which is generally acceptable as well. For most scenarios the match end lookup significantly outweighs the time spent in reversal, except for very dense matches. Of course a notable exception is when the pattern contains [\s\S]*, when you can skip right to the end like Compiled does, but this should be an one-off edge case analyzed during construction.